### PR TITLE
Moved most of the code to openzwave-stateful-rust and adapted the rem…

### PR DESCRIPTION
**Pull request by julienw**
…aining code to make it work

Moved most of the code to https://github.com/fxbox/openzwave-stateful-rust

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -1,171 +1,17 @@
...
-impl Program {
-    pub fn new() -> Program {
-        Program {
-            state: Arc::new(Mutex::new(ProgramState::new()))
-        }
-    }
-}
+extern crate openzwave_stateful as openzwave;
+use openzwave::InitOptions;
+use openzwave::{ ValueGenre, ValueID };
```
Comment (julienw) [inline src/main.rs:3]: I expose these at the top level in this new crate.

File: src/main.rs [hunk 2]
```diff
@@ -1,171 +1,17 @@
...
-
-        println!("found device {}", device);
-
-        match device {
-            "usb" => manager.add_usb_driver(),
-            _ => manager.add_driver(&device)
-        }.unwrap()
-    }
+    let options = InitOptions {
+        device: std::env::args().skip(1).last() // last but not first
```
Comment (julienw) [inline src/main.rs:11]: not completely sure about this yet; maybe we want an Option. Maybe we want to be able to add this later with a function. So this is like it for now.

File: src/main.rs [hunk 3]
```diff
@@ -193,23 +39,23 @@ fn main() {
             "args"                  => println!("args = {:?}", tokens),
             "exit" | "q" | "quit"   => break,
             "controllers"           => {
-                let ref controllers = program.state.lock().unwrap().controllers;
+                let state = ozw.get_state();
```
Comment (julienw) [inline src/main.rs:42]: we need to bind this to a variable so that the borrow checker finds out we borrow this until the end of the block.

**Comment by dhylands** [comment]
r+ - works for me

**Pull request merged**